proxy/src/main/java/com/velocitypowered/proxy/connection/player/resourcepack/LegacyResourcePackHandler.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2024 Velocity Contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.velocitypowered.proxy.connection.player.resourcepack;
+
+import com.velocitypowered.api.event.player.PlayerResourcePackStatusEvent;
+import com.velocitypowered.api.network.ProtocolVersion;
+import com.velocitypowered.api.proxy.player.ResourcePackInfo;
+import com.velocitypowered.proxy.VelocityServer;
+import com.velocitypowered.proxy.connection.client.ConnectedPlayer;
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.List;
+import java.util.Queue;
+import java.util.UUID;
+import net.kyori.adventure.text.Component;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Legacy (Minecraft <1.17) ResourcePackHandler.
+ */
+public sealed class LegacyResourcePackHandler extends ResourcePackHandler
+        permits Legacy117ResourcePackHandler {
+  protected @MonotonicNonNull Boolean previousResourceResponse;
+  protected final Queue<ResourcePackInfo> outstandingResourcePacks = new ArrayDeque<>();
+  private @Nullable ResourcePackInfo pendingResourcePack;
+  private @Nullable ResourcePackInfo appliedResourcePack;
+
+  LegacyResourcePackHandler(final ConnectedPlayer player, final VelocityServer server) {
+    super(player, server);
+  }
+
+  @Override
+  @Nullable
+  public ResourcePackInfo getFirstAppliedPack() {
+    return appliedResourcePack;
+  }
+
+  @Override
+  @Nullable
+  public ResourcePackInfo getFirstPendingPack() {
+    return pendingResourcePack;
+  }
+
+  @Override
+  public @NotNull Collection<ResourcePackInfo> getAppliedResourcePacks() {
+    if (appliedResourcePack == null) {
+      return List.of();
+    }
+    return List.of(appliedResourcePack);
+  }
+
+  @Override
+  public @NotNull Collection<ResourcePackInfo> getPendingResourcePacks() {
+    if (pendingResourcePack == null) {
+      return List.of();
+    }
+    return List.of(pendingResourcePack);
+  }
+
+  @Override
+  public void clearAppliedResourcePacks() {
+    this.appliedResourcePack = null;
+  }
+
+  @Override
+  public boolean remove(final @NotNull UUID id) {
+    if (appliedResourcePack != null && id.equals(appliedResourcePack.getId())) {
```

Comment:
Legacy doesn't support removing a packet from the client, and so I don't think that this is a good idea.

I think that there was some questions over if Velocity wanted to provide some hosted empty packs for the sake of offering them as an API thing, no idea if there was anything against that in the org, but, maybe worth considering if we did want to try to "support" this